src/main/resources/init-script-gradle.ftl

```diff
@@ -200,11 +202,41 @@ def createTaskOutputDirectory(String outputDirectoryPath) {
 }
 
 def shouldInclude(Set<String> excluded, Set<String> included, String value) {
-    return !excluded.contains(value) && (included.isEmpty() || included.contains(value))
+    return !containsWithWildcard(value, excluded) && (included.isEmpty() || containsWithWildcard(value, included))
 }
 
 def convertStringToSet(String value) {
     return value.tokenize(',').toSet()
 }
+
+def containsWithWildcard(String value, Set<String> tokenSet) {
+    for (String token : tokenSet) {
+        if (match(value, token) {
+            return true
+        }
+    }
+    return tokenSet.contains(value)
+}
+
+def match(String value, String token) {
+    def tokenRegex = wildCardTokenToRegexToken(token)
+    return value.matches(tokenRegex)
+}
+
+def wildCardTokenToRegexToken(String token) {
+    Matcher matcher = Pattern.compile(/[^*?]+|(\*)|(\?)/).matcher(token)
+    StringBuffer buffer= new StringBuffer()
+    while (matcher.find()) {
+        if(matcher.group(1) != null) {
+            matcher.appendReplacement(buffer, '.*')
+        } else if (matcher.group(2) != null) {
+            matcher.appendReplacement(buffer, "."); 
+        } else {
+            matcher.appendReplacement(buffer, '\\\\Q' + matcher.group(0) + '\\\\E')
+        }
+    }
+    matcher.appendTail(buffer)
+    return buffer.toString()
+}
```

Comment:
Any way we could include some unit tests for this? While I realize this is a build script this seems to be sophisticated enough to have some tests.

Reply:
That is a valid point. Since, we do not have a precedent in Detect to unit test Groovy script that I am aware of, I performed the unit tests as a sample to validate until we could design one as tech debt:
<img width="2338" alt="Screenshot 2023-08-31 at 10 06 07 AM" src="https://github.com/blackducksoftware/synopsys-detect/assets/63407765/63cf9f63-2ffc-4763-a2a4-65eebbd4bc2e">